packages/theme/src/client/modules/info/components/TOC.ts

```diff
@@ -134,13 +135,39 @@ export default defineComponent({
       );
     });
 
+    const updateTocMarker = (): void => {
+      if (!toc.value) return;
+      const tocTop = toc.value.getBoundingClientRect().top;
+      const tocScrollTop = toc.value.scrollTop;
+
+      const activeTocItem = document.querySelector(".toc-item.active");
+
+      if (!activeTocItem) return;
+      const activeTocItemTop = activeTocItem.getBoundingClientRect().top;
+
+      const tocMarker = document.querySelector(".toc-marker") as HTMLElement;
+
+      tocMarker.style.top = `${activeTocItemTop - tocTop + tocScrollTop}px`;
+    };
+
+    onUpdated(() => {
+      updateTocMarker();
+    });
+
     return (): VNode | null => {
       const tocHeaders = props.items.length
         ? renderChildren(props.items, props.headerDepth)
         : page.value.headers
         ? renderChildren(page.value.headers, props.headerDepth)
         : null;
 
+      (tocHeaders?.children as VNode[] | null)?.push(
+        h("div", {
+          class: "toc-marker",
+          id: "toc-marker", // ensure transitions
+        })
+      );
+
```

Comment:
My question is why not place this below in the returned result?

Reply:
I think this element is better to be a child of `.toc-list`, same as `.toc-list::before`. The css is more consistent in this way. I can't place it in `renderChildren()` since the function is recursive. I also can't place it below in the return statement since it would be `.toc-wrapper`'s child.

Transition won't work without `id: "toc-marker"`. I guess this prop tells vue to trigger transition properly like the `key` attribute?